src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/CommonRules/PosBlockSignatureRuleTest.cs

```diff
@@ -223,7 +223,7 @@ public async Task RunAsync_ProofOfStakeBlock_ScriptKeyDoesNotPassBlockSignatureV
             this.ruleContext.ValidationContext.Block = block;
             Assert.True(BlockStake.IsProofOfStake(this.ruleContext.ValidationContext.Block));
 
-            ConsensusErrorException exception = await Assert.ThrowsAsync<ConsensusErrorException>(() => this.consensusRules.RegisterRule<PosBlockSignatureRule>().RunAsync(this.ruleContext));
+            ConsensusErrorException exception = Assert.Throws<ConsensusErrorException>(() => this.consensusRules.RegisterRule<PosBlockSignatureRule>().Run(this.ruleContext));
```

Comment:
The above tests are asserting the same bad block consensus exception.   

Maybe move the same code into a function.

```
private void AssertBadBlockSignatureWithConsensusException()
{
    Assert.True(BlockStake.IsProofOfStake(this.ruleContext.ValidationContext.Block));

    ConsensusErrorException exception = Assert.Throws<ConsensusErrorException>(() => this.consensusRules.RegisterRule<PosBlockSignatureRule>().Run(this.ruleContext));

    Assert.Equal(ConsensusErrors.BadBlockSignature, exception.ConsensusError);
}
```

Reply:
Totally don't mind. Just outside of the scope of this PR